packages/app/src/overview/components/overview-screens/helpers/use-update-flow-state.hook.tsx

```diff
@@ -3,84 +3,125 @@
       ) {
         silentCheckForUpdatePromise.abort()
         setSilentCheckForUpdatePromise(undefined)
       }
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [silentCheckForUpdateState, forceUpdateNeeded, osVersion])
+  }, [silentCheckForUpdate, needsForceUpdate, osVersion])
 
   useEffect(() => {
-    if (forceUpdateNeeded) {
+    if (needsForceUpdate) {
       return
     }
     if (
-      (silentCheckForUpdateState === SilentCheckForUpdateState.Skipped &&
-        checkingForUpdateState === CheckForUpdateState.Initial) ||
-      silentCheckForUpdateState === SilentCheckForUpdateState.Failed ||
-      checkingForUpdateState === CheckForUpdateState.Failed ||
-      checkingForUpdateState === CheckForUpdateState.PerformedWithFailure
+      silentCheckForUpdate === SilentCheckForUpdateState.Loading ||
+      checkForUpdateState === CheckForUpdateState.Loading ||
+      (checkForUpdateState === CheckForUpdateState.Initial &&
+        silentCheckForUpdate === SilentCheckForUpdateState.Initial)
     ) {
-      setCheckForUpdateLocalState(CheckForUpdateLocalState.Failed)
+      setCheckForUpdateLocalState(CheckForUpdateLocalState.Loading)
+    } else if (allDownloaded) {
+      setCheckForUpdateLocalState(CheckForUpdateLocalState.Install)
+    } else if ((availableReleasesForUpdate || []).length > 0) {
+      setCheckForUpdateLocalState(CheckForUpdateLocalState.Download)
     } else if (
-      silentCheckForUpdateState === SilentCheckForUpdateState.Loading
+      error ||
+      [
+        CheckForUpdateState.PerformedWithFailure,
+        CheckForUpdateState.Failed,
+      ].includes(checkForUpdateState) ||
+      (checkForUpdateState === CheckForUpdateState.Initial &&
+        [
+          SilentCheckForUpdateState.Skipped,
+          SilentCheckForUpdateState.Failed,
+        ].includes(silentCheckForUpdate))
     ) {
-      setCheckForUpdateLocalState(CheckForUpdateLocalState.SilentCheckLoading)
-    } else if (checkingForUpdateState === CheckForUpdateState.Loading) {
-      setCheckForUpdateLocalState(CheckForUpdateLocalState.Loading)
+      setCheckForUpdateLocalState(CheckForUpdateLocalState.Failed)
     } else if (
-      silentCheckForUpdateState === SilentCheckForUpdateState.Loaded ||
-      checkingForUpdateState === CheckForUpdateState.Loaded ||
-      checkingForUpdateState === CheckForUpdateState.Performed
+      [CheckForUpdateState.Loaded, CheckForUpdateState.Performed].includes(
+        checkForUpdateState
+      ) ||
+      SilentCheckForUpdateState.Loaded === silentCheckForUpdate
     ) {
       setCheckForUpdateLocalState(CheckForUpdateLocalState.Loaded)
     }
-  }, [silentCheckForUpdateState, checkingForUpdateState, forceUpdateNeeded])
+  }, [
+    silentCheckForUpdate,
+    checkForUpdateState,
+    needsForceUpdate,
+    allDownloaded,
+    availableReleasesForUpdate,
+    error,
+  ])
   return {
     checkForUpdateLocalState,
   }
```

Comment:
is it necessary to return object with one field? :)

Reply:
I was thinking that maybe a good idea would be to return a set of booleans representing those statuses. What do you think?